charon/src/translate_types.rs

```diff
@@ -506,10 +507,19 @@ impl<'tcx, 'ctx, 'ctx1> BodyTransCtx<'tcx, 'ctx, 'ctx1> {
     ) -> Result<TypeDeclKind, Error> {
         use rustc_middle::ty::AdtKind;
         let def_span = self.t_ctx.tcx.def_span(rust_id);
-        // Don't use `hax::AdtDef` because it loses `VariantIdx` information.
-        let adt: rustc_middle::ty::AdtDef = self.t_ctx.tcx.adt_def(rust_id);
         trace!("{}", trans_id);
 
+        let is_foreign = rust_id.as_local().is_some_and(|local_def_id| {
+            let node = self.t_ctx.tcx.hir().find_by_def_id(local_def_id);
+            matches!(node, Some(HirNode::ForeignItem(_)))
+        });
```

Comment:
Factor out this?

Reply:
lol yeah sorry x)

Reply:
Could we include this in the `id_is_transparent` check?

Reply:
Yeah that's much nicer